packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -71,164 +77,250 @@
   void onError(String errorCode, String errorMessage);
 }
 
-public class Camera {
+/**
+ * Note: at this time we do not implement zero shutter lag (ZSL) capture. This is a potential
+ * improvement we can use in the future. The idea is in a TEMPLATE_ZERO_SHUTTER_LAG capture session,
+ * the system maintains a ring buffer of images from the preview. It must be in full auto moved
+ * (flash, ae, focus, etc). When you capture an image, it simply picks one out of the ring buffer,
+ * thus capturing an image with zero shutter lag.
+ *
+ * <p>This is a potential improvement for the future. A good example is the AOSP camera here:
```

Comment:
what is a potential improvement for the future?  can this be a comment in the close to the implementation?

Reply:
@acoutts same here

Reply:
I'm trying to remember my thought process here.

There is always a shutter lag in our camera implementation. I did a lot of debug printing and ultimately found the delay is unavoidable and happens between when we submit our capture request and when the request actually starts, so I suspect it's the camera allocating memory and getting ready to handle the request.

The api offers a zero shutter lag (ZSL) request which is like a different mode for the preview, where there's a ring buffer constantly capturing frames and then when you "submit a request", it just picks off the last one and saves it, so there's no delay warming up the camera.

Not all hardware supports this, so it would not be a small task to implement correctly, but it's available to us in the camera2 api.